atomic-copy/atomic-copy-safe.sh

```diff
@@ -0,0 +1,20 @@
+#! @bash@/bin/sh -e
+
+# copy+paste of copyToKernelsDir https://github.com/NixOS/nixpkgs/blob/904ecf0b4e055dc465f5ae6574be2af8cc25dec3/nixos/modules/system/boot/loader/generic-extlinux-compatible/extlinux-conf-builder.sh#L53
+
+shopt -s nullglob
+
+export PATH=/empty
+for i in @path@; do PATH=$PATH:$i/bin; done
+
+src=$(readlink -f "$1")
+dst="$2"
+
+# Don't copy the file if $dst already exists.
+# Also create $dst atomically to prevent partially copied files
+# if this script is ever interrupted.
+if ! test -e $dst; then
+	dstTmp=$dst.tmp.$$
+	cp -r $src $dstTmp
+	mv $dstTmp $dst
```

Comment:
@tstat you said:

> Additionally, the systemd script that installs the kernel, initrd, and firmware today has some care taken to not modify any files that haven't changed, and to track if the installation process was halted at a bad time (like in the middle of firmware installation). It would be good to retain that.

This line moves the files into the right place after they've been copied to the boot filesystem. Notice line 16 though - the file won't be copied if it already exists at the destination. This is copied from the `generic-extlinux-compatible` shell script. A modified version `atomic-copy-clobber.sh` does the same but overwrites the file, which is used for `config.txt` and `cmdline.txt`.  Is this in line with what you mean?